### ISSUE TITLE
Draft Mode Special Cases

### ISSUE THREAD

**Issue by vicchig** [MajeedKazemi/nova-editor#181]
In these cases going into the draft mode is currently possible, but will not yield any valid results because it is impossible to manipulate these expressions and make them valid. Should we not enter the draft mode in these cases?

- Inserting a boolean into an arithmetic expression. Ex. (True + ---)
- Inserting range() into an arithmetic expression. Ex. (range(1, 2) - ---). 
- Inserting anything that is a Number into a hole that expects a list or string. **This case is fine because we can always do <code>[number]</code> or <code>str(number)</code>**

In short, anywhere where a type conversion is simply impossible we should not enter the draft mode. **The question that remains is how do we recognize cases where a type insertion is not possible.**

**Comment by vicchig** [comment]
We have code to address this. Not sure if all cases are caught, but they should be. See TypeConversionMap in util.ts
